src/ui/ons/ONSTextField.svelte

```diff
@@ -1,15 +1,56 @@
 <script>
-  export let id;
-  export let value = "";
+  export let id, type, textFieldValue, onChange;
   export let hint = "";
+  export let inputLabel = "";
+  export let showPlaceholder = false;
+  export let errorMessage = "Custom error message";
+  export let renderErrorMsg = false;
+
+  function typeAction(node) {
+    node.type = type;
+  }
 </script>
 
-<div class="ons-field">
-  <label class="ons-label" for={id}><slot /></label>
-  {#if hint}
-    <span id="description-hint" class="ons-label__description  ons-input--with-description">
-      {hint}
-    </span>
-  {/if}
-  <input type="text" {id} class="ons-input ons-input--text ons-input-type__input" bind:value />
-</div>
+{#if renderErrorMsg}
+  <div class="ons-panel ons-panel--error ons-panel--no-title ons-u-mb-s" id="{id}-error">
+    <span class="ons-u-vh">Error: </span>
+    <div class="ons-panel__body">
+      <p class="ons-panel__error">
+        <strong>{errorMessage}</strong>
+      </p>
+      <div class="ons-field">
+        <label class="ons-label {showPlaceholder ? 'ons-label--placeholder' : ''}" for={id}>{inputLabel}</label>
+        {#if hint}
+          <span id="description-hint" class="ons-label__description  ons-input--with-description">
+            {hint}
+          </span>
+        {/if}
+        <input
+          {id}
+          class="ons-input ons-input--text ons-input-type__input {showPlaceholder ? 'ons-input--placeholder' : ''}"
+          placeholder={showPlaceholder ? inputLabel : ""}
```

Comment:
I think placeholders can be any text - unless there is something I'm missing
Make `placeholder` a text variable and remove `showPlaceholder`

Reply:
Because I wanted to create an accessible component I had a look at the only example with placeholder I was able to find in the ONS DS components and I took the search component (2nd example) as a reference : https://ons-design-system.netlify.app/components/search/ . And in that example there are 2 classes one on the label >> .ons-label--placeholder ==> this is forcing whatever is passed down as a label to display as a placeholder;  and the second class on the actual input >> .ons-input--placeholder ==> this class is basically hiding whatever placeholder I am passing down as a text variable. Now, should I remove this 2 ons classes  so that I can pass down the placeholder as text variable ?

Reply:
@thomasridd , as discussed this morning I kept the placeholder as it was before you added the comment in and the only change I made was to rename the showPlaceholder prop/bool to accessiblePlaceholder as per the ONS DS pattern.